src/cli/services/__tests__/linter.test.ts

```diff
@@ -0,0 +1,386 @@
+import { resolve } from '@stoplight/path';
+import * as nock from 'nock';
+import * as yargs from 'yargs';
+import { ValidationError } from '../../../rulesets/validation';
+import { ILintConfig } from '../../../types/config';
+import lintCommand from '../../commands/lint';
+import { lint } from '../linter';
+
+jest.mock('../output');
+
+const oas2PetstoreSpecPath = resolve(__dirname, '../../../__tests__/__fixtures__/petstore.oas2.json');
+const validCustomOas3SpecPath = resolve(__dirname, '__fixtures__/openapi-3.0-valid-custom.yaml');
+const invalidRulesetPath = resolve(__dirname, '__fixtures__/ruleset-invalid.yaml');
+const validRulesetPath = resolve(__dirname, '__fixtures__/ruleset-valid.yaml');
+const validNestedRulesetPath = resolve(__dirname, '__fixtures__/ruleset-extends-valid.yaml');
+const invalidNestedRulesetPath = resolve(__dirname, '__fixtures__/ruleset-extends-invalid.yaml');
+const standardOas3RulesetPath = resolve(__dirname, '../../../rulesets/oas3/index.json');
+const standardOas2RulesetPath = resolve(__dirname, '../../../rulesets/oas2/index.json');
+const draftRefSpec = resolve(__dirname, './__fixtures__/draft-ref.oas2.json');
+const draftNestedRefSpec = resolve(__dirname, './__fixtures__/draft-nested-ref.oas2.json');
+const validOas3SpecPath = resolve(__dirname, './__fixtures__/openapi-3.0-valid.yaml');
+const invalidOas3SpecPath = resolve(__dirname, '__fixtures__/openapi-3.0-no-contact.yaml');
+
+function run(command: string) {
+  const parser = yargs.command(lintCommand).help();
+  const { document, ruleset, ...opts } = (parser.parse(command) as unknown) as ILintConfig & { document: string };
+  return lint(document, opts, ruleset);
+}
+
+describe('Linter service', () => {
+  let logSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    logSpy = jest.spyOn(console, 'log');
+  });
+
+  afterEach(() => {
+    logSpy.mockRestore();
+
+    nock.cleanAll();
+  });
+
+  it('should handle relative path to a document', async () => {
+    const results = await run('lint src/__tests__/__fixtures__/gh-474/spec.yaml');
+    expect(results).toEqual(
+      expect.not.arrayContaining([
+        expect.objectContaining({
+          code: 'invalid-ref',
+        }),
+      ]),
+    );
+
+    expect(results).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          code: 'oas3-schema',
+          range: {
+            end: {
+              character: 14,
+              line: 5,
+            },
+            start: {
+              character: 10,
+              line: 5,
+            },
+          },
+          source: expect.stringContaining('/__tests__/__fixtures__/gh-474/common.yaml'),
+        }),
+      ]),
+    );
+  });
+
+  it('should handle relative path to a document #2', async () => {
```

Comment:
do we need 2 of `should handle relative path to a document`?

Reply:
Yeah, there are 2 types of paths there. Can you think of any other naming here? :D